src/Hint/NewType.hs

```diff
@@ -65,3 +77,44 @@ hasNoStrategy _                      = False
 withNewtype :: Deriving a -> Deriving a
 withNewtype (Deriving l Nothing rs)  = Deriving l (Just $ DerivNewtype l) rs
 withNewtype d                        = d
+
+data WarnNewtype = WarnNewtype
```

Comment:
I prefer to have a datatype with named fields for better documentation, instead of returning a tuple. Is this fine?

Reply:
Personal preference - I would probably have opted for a tuple, but it's the kind of thing I consider whoever writing the code gets to do whatever they prefer. Neither is better or worse, so happy to go with your approach.